src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySinkConfig.java

```diff
@@ -172,6 +187,79 @@ void validateCmekKey(FailureCollector failureCollector, Map<String, String> argu
     validateCmekKeyLocation(cmekKeyName, null, location, failureCollector);
   }
 
+  /**
+   * Validates job label key value pairs, as per the following rules:
+   * Keys and values can contain only lowercase letters, numeric characters, underscores, and dashes.
+   * Defined in the following link:
+   * <a href="https://cloud.google.com/bigquery/docs/adding-labels#job-label">Docs</a>
+   * @param failureCollector failure collector
+   */
+  void validateJobLabelKeyValue(FailureCollector failureCollector) {
+
+    Set<String> reservedKeys = ImmutableSet.of("job_source", "type");
+
+    String validLabelRegex = "[a-z0-9_-]+";
+    String labelKeyValue = getJobLabelKeyValue();
+
+    if (Strings.isNullOrEmpty(labelKeyValue)) {
+      return;
+    }
+
+    String[] keyValuePairs = labelKeyValue.split(",");
+    Set<String> uniqueKeys = new HashSet<>();
+
+    for (String keyValuePair : keyValuePairs) {
+
+      String[] keyValue = keyValuePair.trim().split(":");
+      boolean isKeyPresent = keyValue.length == 1 || keyValue.length == 2;
```

Comment:
Are empty values allowed i.e. `keyValue.length == 1`? Have we tested this case?

Reply:
Yes, this has been tested.
It's mentioned in the docs [here](https://cloud.google.com/bigquery/docs/adding-labels#adding_a_label_without_a_value).

Reply:
Thanks for the clarification, can you please link this doc as a comment in the code?

Reply:
This comment is still not resolved.

Reply:
>// Adding a label without a value is valid behavior
>// Read more here: https://cloud.google.com/bigquery/docs/adding-labels#adding_a_label_without_a_value